integrations/ttscache-s3/src/S3TtsCache.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  S3Client,
+  PutObjectCommand,
+  PutObjectCommandInput,
+  HeadObjectCommand,
+  HeadObjectCommandInput,
+  GetObjectCommand,
+  GetObjectCommandInput,
+  GetObjectCommandOutput,
+  S3ClientConfig
+} from '@aws-sdk/client-s3';
+
+import { Readable } from 'stream';
+
+import { urlJoin } from 'url-join-ts';
+
+import { TtsCachePlugin, TtsCachePluginConfig, TtsData, AudioUtilities, RequiredOnlyWhere } from '@jovotech/framework';
+
+export interface S3TtsCacheConfig extends TtsCachePluginConfig {
+  bucket: string;
+  path: string;
+  baseUrl: string;
```

Comment:
Hey @rmtuckerphx, @aswetlow and I are finishing up the review and have one last question regarding `baseUrl`:

In your example, you have `bucket` as `mybucket-public` and `baseUrl` as `https://mybucket-public.s3.amazonaws.com`.

Does the URL always look like this format? `https://<bucket>.s3.amazonaws.com`
If yes, we could maybe remove `baseUrl` and put the string together ourselves.

What do you think?

Reply:
@jankoenig I know in the past that the url format changed for newer buckets, so I didn't want to tie these too closely.

Reply:
@jankoenig @aswetlow  I dug up this old article that explained the transition of bucket urls:
https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/

Reading this again, I think that going forward we only need to support virtual-hosted style URL so we can remove `baseUrl` from config and construct it in code.

Thanks for adding `libraryConfig` to support `S3ClientConfig`. That is the better approach.